,dep,version,filepath,bloated,cve,affected versions,advisory
0,tqdm,>=4.26.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,no one,no one,no one
1,matplotlib,>=3.0.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,no one,no one,no one
2,numpy,>=1.15,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2019-6446,<1.16.3,"Numpy 1.16.3 includes a fix for CVE-2019-6446: It uses the pickle Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a numpy.load call.
https://github.com/numpy/numpy/commit/89b688732b37616c9d26623f81aaee1703c30ffb"
3,numpy,>=1.15,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2021-33430,<1.21.0rc1,"Numpy 1.21.0rc1 includes a fix for CVE-2021-33430: A Buffer Overflow vulnerability in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. 
NOTE: The vendor does not agree this is a vulnerability. In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.
https://github.com/numpy/numpy/issues/18939"
4,numpy,>=1.15,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2021-34141,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-34141: An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. 
NOTE: the vendor states that this reported code behavior is ""completely harmless.""
https://github.com/numpy/numpy/issues/18993"
5,numpy,>=1.15,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2021-41496,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-41496: Buffer overflow in the array_from_pyobj function of fortranobject.c, which allows attackers to conduct a Denial of Service attacks by carefully constructing an array with negative values. 
NOTE: The vendor does not agree this is a vulnerability; the negative dimensions can only be created by an already privileged user (or internally).
https://github.com/numpy/numpy/issues/19000"
6,numpy,>=1.15,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2021-41495,<1.22.2,"Numpy 1.22.2  includes a fix for CVE-2021-41495: Null Pointer Dereference vulnerability exists in numpy.sort in NumPy in the PyArray_DescrNew function due to missing return-value validation, which allows attackers to conduct DoS attacks by repetitively creating sort arrays. 
NOTE: While correct that validation is missing, an error can only occur due to an exhaustion of memory. If the user can exhaust memory, they are already privileged. Further, it should be practically impossible to construct an attack which can target the memory exhaustion to occur at exactly this place.
https://github.com/numpy/numpy/issues/19038"
7,pandas,>=0.25,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2020-13091,">=0,<1.0.4","** DISPUTED ** pandas through 1.0.3 can unserialize and execute commands from an untrusted file that is passed to the read_pickle() function, if __reduce__ makes an os.system call. NOTE: third parties dispute this issue because the read_pickle() function is documented as unsafe and it is the user's responsibility to use the function in a secure manner."
8,scipy,>=1.1.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2013-4251,">=0,<0.12.1",The scipy.weave component in SciPy before 0.12.1 creates insecure temporary directories.
9,torch,>=1.1.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,CVE-2022-45907,">=0,<1.13.1","Torch 1.13.1 includes a fix for CVE-2022-45907: In PyTorch before trunk/89695, torch.jit.annotations.parse_type_line can cause arbitrary code execution because eval is used unsafely.
https://github.com/pytorch/pytorch/issues/88868"
10,fcd_torch,>=1.0.5,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,no one,no one,no one
11,seaborn,>=0.9.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,no one,no one,no one
12,pomegranate,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/moses/setup.py,False,no one,no one,no one
